tests/Doctrine/Tests/DBAL/Functional/Platform/DefaultExpressionTest.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\DBAL\Functional\Platform;
+
+use Doctrine\DBAL\FetchMode;
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Platforms\MySqlPlatform;
+use Doctrine\DBAL\Platforms\OraclePlatform;
+use Doctrine\DBAL\Schema\Table;
+use Doctrine\DBAL\Types\Types;
+use Doctrine\Tests\DbalFunctionalTestCase;
+use function sprintf;
+
+class DefaultExpressionTest extends DbalFunctionalTestCase
+{
+    /**
+     * @dataProvider expressionProvider
+     */
+    public function testDefaultExpression(string $type, callable $expression) : void
+    {
+        $platform   = $this->connection->getDatabasePlatform();
+        $defaultSql = $expression($platform, $this);
+
+        $table = new Table('default_expr_test');
+        $table->addColumn('actual_value', $type);
+        $table->addColumn('default_value', $type, ['default' => $defaultSql]);
+        $this->connection->getSchemaManager()->dropAndCreateTable($table);
+
+        $query = sprintf(
+            'INSERT INTO default_expr_test (actual_value) VALUES (%s)',
+            $defaultSql
+        );
+
+        $this->connection->exec($query);
+
+        $query                        = sprintf('SELECT default_value, actual_value FROM default_expr_test');
+        [$actualValue, $defaultValue] = $this->connection->query($query)->fetch(FetchMode::NUMERIC);
+
+        self::assertEquals($actualValue, $defaultValue);
+    }
+
+    /**
+     * @return mixed[][]
+     */
+    public static function expressionProvider() : iterable
+    {
+        yield 'current-date' => [
+            Types::DATE_MUTABLE,
+            static function (AbstractPlatform $platform) : string {
+                if ($platform instanceof MySqlPlatform) {
+                    self::markTestSkipped('Not supported on MySQL');
+                }
+
+                return $platform->getCurrentDateSQL();
+            },
+        ];
+
+        yield 'current-time' => [
+            Types::TIME_MUTABLE,
+            static function (AbstractPlatform $platform) : string {
+                if ($platform instanceof MySqlPlatform) {
+                    self::markTestSkipped('Not supported on MySQL');
+                }
+
+                if ($platform instanceof OraclePlatform) {
+                    self::markTestSkipped('Not supported on Oracle');
```

Comment:
Uhh, I don't think we should skip tests from code within the data-provider: confusing approach IMO

Reply:
We can move this test under `AbstractPlatformTest` and split it into three, one for each expression. Then, in each concrete class, skip the corresponding method if the platform doesn't support it.